src/identity-providers/add/DetailSettings.tsx

```diff
@@ -95,11 +98,19 @@ export const DetailSettings = () => {
   const { realm } = useRealm();
 
   useFetch(
-    () => adminClient.identityProviders.findOne({ alias: id }),
+    () => {
+      return Promise.all([
+        adminClient.identityProviders.findOne({ alias: id }),
+        adminClient.identityProviders.findMappers({ alias: id }),
+        adminClient.identityProviders.findMapperTypes({ alias: id }),
+      ]);
+    },
     (provider) => {
       if (provider) {
-        setProvider(provider);
-        Object.entries(provider).map((entry) => setValue(entry[0], entry[1]));
+        setProvider({ ...provider[0] });
+        Object.entries(provider[0]).map((entry) =>
+          setValue(entry[0], entry[1])
+        );
```

Comment:
If you destructure the entry this code can be simplified.
```suggestion
        Object.entries(provider).map(([key, value) =>
          setValue(key], value)
        );
```

I think you can even use `reset()` instead.
```suggestion
        reset(provider);
```